.github/workflows/test_comprehensive.yml

```diff
@@ -35,10 +35,16 @@ jobs:
       - name: Check Manifest
         run: check-manifest
 
+  build_wheel:
+    name: Build wheel
+    uses: ./.github/workflows/resuable_build_wheel.yml
+
   test:
-    name: ${{ matrix.platform }} py${{ matrix.python }} ${{ matrix.toxenv }} ${{ matrix.backend }} ${{ matrix.MIN_REQ && 'min_req' }}
-    timeout-minutes: 60
-    runs-on: ${{ matrix.platform }}
+    name: ${{ matrix.platform }}
+    uses: ./.github/workflows/reusable_run_tox_test.yml
+    needs: build_wheel
+    concurrency:
+      group: comprehensive-test-${{ matrix.platform }}
```

Comment:
This limits tests on the main branch to run only a single test from a given system. 

This is to prevent situations where merging something to main blocks all PR tests.

Reply:
Consider adding this comment to this workflow file itself.